### PR TITLE
add more forms to E24 resource type

### PR DESCRIPTION
**Pull request by mradamcox**
### Description of Change
Adding multiple forms to the HERITAGE_FEATURE.E24 resource type.

- Assessment Summary
    same as used for E27
- Resource Summary
    new form created based on the summary form for E27; summary-feature.htm/summary-feature.js
- Location
    same as used for E27, but slight modification made to conditionally change one node name
- Measurements
    same as used for E27
- Related Resource
    standard form

Note that PR #6 should be merged before this one.

### CODE REVIEW

File: eamena/eamena/media/js/views/forms/location.js [hunk 1]
```diff
@@ -102,10 +102,16 @@ define([
                 }
             }));
             
+            if (resourcetypeid == 'HERITAGE_FEATURE.E24') {
+                var siteshape_node = 'SITE_OVERALL_SHAPE_TYPE.E55';
+            } else { 
+                var siteshape_node = 'SITE_OVERALL_SHAPE_TYPE_NEW.E55';
```
Comment (azerbini) [inline eamena/eamena/media/js/views/forms/location.js:108]: @mradamcox  the _NEW tag will disappear at the end of the data migration process (using the rename_entity_type utility: https://github.com/azerbini/eamena_dev/blob/15561846fab51b0a8dc89aebd3240d8e41482c26/arches/management/commands/package_utils/migrate_resources.py#L385) so this bit of code will become redundant then.

File: eamena/eamena/models/forms.py [hunk 2]
```diff
@@ -854,11 +893,16 @@ def get_info():
...
         self.update_nodes('GEOMETRIC_PLACE_EXPRESSION.SP5', data)
         self.update_nodes('GEOMETRY_EXTENT_CERTAINTY.I6', data)
         self.update_nodes('GRID_ID.E42', data)
-        self.update_nodes('SITE_OVERALL_SHAPE_TYPE_NEW.E55', data)
+        
+        if self.resource.entitytypeid == 'HERITAGE_FEATURE.E24':
+            siteshape_node = 'SITE_OVERALL_SHAPE_TYPE.E55'
+        else:
+            siteshape_node = 'SITE_OVERALL_SHAPE_TYPE_NEW.E55'
+        self.update_nodes(siteshape_node, data)
```
Comment (azerbini) [inline eamena/eamena/models/forms.py:927]: @mradamcox  the _NEW tag will disappear at the end of the data migration process (using the rename_entity_type utility: https://github.com/azerbini/eamena_dev/blob/15561846fab51b0a8dc89aebd3240d8e41482c26/arches/management/commands/package_utils/migrate_resources.py#L385) so this bit of code will become redundant then.

File: eamena/eamena/models/forms.py [hunk 3]
```diff
@@ -892,10 +937,15 @@ def load(self, lang):
                 'domains': {}
             }
         
-        self.data['SITE_OVERALL_SHAPE_TYPE_NEW.E55'] = {
-            'branch_lists': self.get_nodes('SITE_OVERALL_SHAPE_TYPE_NEW.E55'),
+        if self.resource.entitytypeid == 'HERITAGE_FEATURE.E24':
+            siteshape_node = 'SITE_OVERALL_SHAPE_TYPE.E55'
+        else:
+            siteshape_node = 'SITE_OVERALL_SHAPE_TYPE_NEW.E55'
```
Comment (azerbini) [inline eamena/eamena/models/forms.py:966]: @mradamcox  the _NEW tag will disappear at the end of the data migration process (using the rename_entity_type utility: https://github.com/azerbini/eamena_dev/blob/15561846fab51b0a8dc89aebd3240d8e41482c26/arches/management/commands/package_utils/migrate_resources.py#L385) so this bit of code will become redundant then.

File: eamena/eamena/templates/views/forms/location.htm [hunk 4]
```diff
@@ -350,9 +350,15 @@ <h5 id="arches-subjects-form" class="section-headline">{% trans "Overall Site Sh
...
+                                {% else %}
                                 <input style="width:33%; height: 36px; padding-top: 2px;" class="select2 arches-select2-crud-form resource-type hidden-xs" id="culturalp-dropdown" placeholder="{% trans 'Shape' %}" data-bind="select2: {value: getEditedNode('SITE_OVERALL_SHAPE_TYPE_NEW.E55', 'value'), dataKey: 'SITE_OVERALL_SHAPE_TYPE_NEW.E55'}"></input>
                                 
                                 <input style="width:33%; height: 36px; padding-top: 2px;" class="select2 arches-select2-crud-form resource-type  hidden-sm hidden-md hidden-lg" id="culperiod-dropdown" placeholder= "{% trans 'Shape' %}" data-bind="select2: {value: getEditedNode('SITE_OVERALL_SHAPE_TYPE_NEW.E55', 'value'), dataKey: 'SITE_OVERALL_SHAPE_TYPE_NEW.E55'}"></input>
+                                {% endif %}
```
Comment (azerbini) [inline eamena/eamena/templates/views/forms/location.htm:361]: @mradamcox  the _NEW tag will disappear at the end of the data migration process (using the rename_entity_type utility: https://github.com/azerbini/eamena_dev/blob/15561846fab51b0a8dc89aebd3240d8e41482c26/arches/management/commands/package_utils/migrate_resources.py#L385) so this bit of code will become redundant then.

File: eamena/eamena/templates/views/forms/location.htm [hunk 5]
```diff
@@ -387,7 +393,12 @@ <h5 id="arches-subjects-form" class="section-headline">{% trans "Overall Site Sh
...
-                                <a href="javascript:void(0)" class="arches-CRUD-child arches-CRUD-edit" data-bind="click: $parent.editItem.bind($parent)" title="{% trans "Edit this value" %}"><span data-bind='{text: nodes.get("SITE_OVERALL_SHAPE_TYPE_NEW.E55", "label")}' style="padding-right:2px;"></span>
+                                <a href="javascript:void(0)" class="arches-CRUD-child arches-CRUD-edit" data-bind="click: $parent.editItem.bind($parent)" title="{% trans 'Edit this value' %}">
+                                {% if resourcetypeid == 'HERITAGE_FEATURE.E24' %}
+                                <span data-bind='{text: nodes.get("SITE_OVERALL_SHAPE_TYPE.E55", "label")}' style="padding-right:2px;"></span>
+                                {% else %}
+                                <span data-bind='{text: nodes.get("SITE_OVERALL_SHAPE_TYPE_NEW.E55", "label")}' style="padding-right:2px;"></span>
+                                {% endif %}
```
Comment (azerbini) [inline eamena/eamena/templates/views/forms/location.htm:401]: @mradamcox  the _NEW tag will disappear at the end of the data migration process (using the rename_entity_type utility: https://github.com/azerbini/eamena_dev/blob/15561846fab51b0a8dc89aebd3240d8e41482c26/arches/management/commands/package_utils/migrate_resources.py#L385) so this bit of code will become redundant then.

**Comment by mradamcox** [comment]
@azerbini ok, in that case those if statements will need to be removed after the migration process is complete. They will cause errors.

File: eamena/eamena/models/forms.py [hunk 6]
```diff
@@ -114,6 +114,45 @@ def load(self, lang):
...
                 }
             }
+            
+class FeatureSummaryForm(ResourceForm):
+    @staticmethod
+    def get_info():
+        return {
+            'id': 'summary-feature',
+            'icon': 'fa-align-center',
+            'name': _('Resource Summary'),
```
Comment (azerbini) [inline eamena/eamena/models/forms.py:124]: @mradamcox Resource Summary should be called 'Feature Summary' instead at the Feature level.

File: eamena/eamena/models/forms.py [hunk 7]
```diff
@@ -114,6 +114,45 @@ def load(self, lang):
...
+            'class': FeatureSummaryForm
+        }
+
+    def update(self, data, files):
+        self.update_nodes('NAME.E41', data)
+        self.update_nodes('HERITAGE_CLASSIFICATION_TYPE.E55', data)
```
Comment (azerbini) [inline eamena/eamena/models/forms.py:130]: A node is missing here, `HERITAGE_FEATURE_USE_TYPE.E55`. Could you please add it to this form, @mradamcox ?

**Pull request merged**